plugins/janus_videoroom.c

```diff
@@ -944,21 +969,19 @@ static void janus_videoroom_notify_participants(janus_videoroom_publisher *parti
 }
 
 static void janus_videoroom_leave_or_unpublish(janus_videoroom_publisher *participant, gboolean is_leaving) {
+	/* we need to check if the room still exists, may have been destroyed already */
+	if(!participant->room || participant->room->destroyed)
```

Comment:
Shouldn't this destroyed check be atomic? I noticed I still left some non-atomic checks of destroyed myself, which is wrong, as we don't use the destroyed value as a trigger for lazy frees anymore in this branch.